### PR TITLE
Server groups should be disabled prior to being destroyed

### PR DESCRIPTION
**Pull request by ajordens**
- Insert a DisableClusterStage prior to any ShrinkClusterStage
- WaitForClusterDisableTask will wait a particular amount of time to allow for any connection draining / discovery out_of_service to complete

### CODE REVIEW

**Comment by ajordens** [comment]
Thoughts on this approach @cfieber?

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/WaitForClusterDisableTask.groovy [hunk 1]
```diff
@@ -16,14 +16,33 @@
...
 import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.support.TargetServerGroup
 import com.netflix.spinnaker.orca.clouddriver.utils.HealthHelper
+import com.netflix.spinnaker.orca.pipeline.model.Stage
 import org.springframework.stereotype.Component
 
 import java.util.function.Function
 
 @Component
 class WaitForClusterDisableTask extends AbstractWaitForClusterWideClouddriverTask {
+  private static final int MINIMUM_WAIT_TIME_MS = 90000
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/WaitForClusterDisableTask.groovy]: Not a huge fan of the static wait ... but the current behavior has isServerGroupOperationInProgress completing as soon as the server group is disabled (_vs. waiting for health providers to go DOWN_). 

This smells like a bug but at the very least a small wait should account for any connection draining / discovery activity to complete.

**Comment by robfletcher** [comment]
So the disable cluster will affect the exact same ASGs that the shrink will? Is there any chance of another operation interfering?

**Comment by ajordens** [comment]
I don't believe another operation should interfere ... these stages do look for and ignore any server groups that are newer than the current pipeline.

**Comment by robfletcher** [comment]
ok, in that case LGTM

**Pull request merged**